modules/oauth/src/main/java/org/mule/extension/oauth2/internal/AbstractTokenRequestHandler.java

```diff
@@ -114,13 +161,19 @@ protected Event invokeTokenUrl(final Event event) throws MuleException, TokenUrl
                       null);
 
       Message responseMessage = httpResponseToMuleMessage.convert(event, response, tokenUrl);
-      Event responseEvent = Event.builder(event).message(InternalMessage.builder(responseMessage).build()).build();
+      final Builder responseEventBuilder = Event.builder(event).message((InternalMessage) responseMessage);
 
       if (((HttpResponseAttributes) responseMessage.getAttributes()).getStatusCode() >= BAD_REQUEST.getStatusCode()) {
-        throw new TokenUrlResponseException(responseEvent);
+        throw new TokenUrlResponseException(responseEventBuilder.build());
       }
 
-      return responseEvent;
+      if (responseMessage.getPayload().getDataType().isStreamType()) {
+        return responseEventBuilder.message(InternalMessage.builder(responseMessage)
```

Comment:
no extension should be doing things like this, the runtime should. However, @afelisatti will soon decouple the http extension from the mule message so this will probably be fixed then